util/grpcerrors/grpcerrors.go

```diff
@@ -72,6 +72,21 @@ func ToGRPC(ctx context.Context, err error) error {
 	return st.Err()
 }
 
+// errorHasMoreContext checks if the original error provides more context by having
+// a different message or additional details than the Status.
+func errorHasMoreContext(err error, st *status.Status) bool {
+	if errMessage := err.Error(); len(errMessage) > len(st.Message()) {
```

Comment:
what's this length check for?

Reply:
That's a narrower version of the pre-existing check at https://github.com/moby/buildkit/pull/5905/files#diff-ab206f22f3b8cb9ead9e872d93d8a668f181b0bbdc75f29059a88ebbff1abe13L48 - You don't want to replace the original error message if somehow we ended up with a shorter one, because the objective is to keep the maximum amount of context.

The general idea of that function is that if the original error had a longer string than the `status.Status` message, we replace the status message with the full error message unless we can prove that the longer error message only comes from prepending the `codes.Code`.

Reply:
@tonistiigi ping :-)